src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -359,6 +359,26 @@ public async Task FailedConnect_ConnectedReturnsFalse()
 
             Assert.False(socket.Connected);
         }
+
+        [Fact]
+        public async Task FailedConnect_ConnectedReturnsFalse_TimeSpan()
```

Comment:
why not parameterize the existing test - similar to what you did for Allow_Connection_UnderDifferentUsers_ForClientReading ? It would save repeating all this code, wihch then has to be maintained, etc.

BTW nit if you did have it separate -- it would be cleaner to not bother with the timeSpan variable IMO.